notation.go

```diff
@@ -325,3 +326,20 @@ func generateAnnotations(signerInfo *signature.SignerInfo) (map[string]string, e
 		annotationX509ChainThumbprint: string(val),
 	}, nil
 }
+
+// GetDescriptorFromPayload parses a signature payload and returns the descriptor
+// that was signed.
+func GetDescriptorFromPayload(payload *signature.Payload) (*ocispec.Descriptor, error) {
+	if payload == nil {
+		return nil, errors.New("empty payload")
+	}
+
+	var parsedPayload envelope.Payload
+
+	err := json.Unmarshal(payload.Content, &parsedPayload)
```

Comment:
Security concern: payload media type is not checked before parsing.

Reply:
can parse the payload to an intermediate struct to check mediatype before parsing to the final payload struct.